### PR TITLE
Clarify writable behavior of readonly-named `Permissions` methods.

### LINKED ISSUE #41984: fs::Permissions set_readonly is odd on Unix.
- `set_readonly(true)` wipes out write permissions, leaving existing executable permissions in place. This seems contrary to what I would expect this function to do: make something _only_ readable. I would expect `set_readonly(true)` to set the permissions to `0o444`.

- `set_readonly(false)` _adds_ write permissions to make permissions _not_ readonly, as opposed to wiping out read permissions. Setting permissions to not readonly itself is a bit odd to begin with. Any combination of permissions that are not just read is technically not read only. I don't have a good solution to this, but I think more clarifying documentation would suffice.

### PR DESCRIPTION
**Pull request by frewsxcv**
Opened primarily to fix https://github.com/rust-lang/rust/issues/41984.

### CODE REVIEW

File: src/libstd/fs.rs [hunk 1]
```diff
@@ -934,7 +934,10 @@ impl Permissions {
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn readonly(&self) -> bool { self.0.readonly() }
 
-    /// Modifies the readonly flag for this set of permissions.
+    /// Modifies the readonly flag for this set of permissions. If the
+    /// `readonly` argument is `true`, using the resulting `Permission` will
+    /// make the file unwritable. Conversely, if it's is `false`, using the
+    /// resulting `Permission` will make the file writable.
```
Comment (frewsxcv) [inline src/libstd/fs.rs]: This all feels pretty wordy, if anyone has suggestions to improve the flow here, would love to hear!

**Comment by QuietMisdreavus** [comment]
copying from our conversation in IRC:

"Unwritable", while accurate, sounds a little clumsy. The `readonly` function can keep it as-is, but the `set_readonly` might work better by breaking it out:

> Modifies the readonly flag for this set of permissions. If the
> `readonly` argument is `true`, using the resulting `Permission` will
> forbid writing to the file. Conversely, if it's is `false`, using the
> resulting `Permission` will allow writing to the file.

I originally had "will make the file unable to be written to" for the "if true" sentence, but i kinda like the allow/forbid dichotomy like this.

**Comment by frewsxcv** [comment]
@QuietMisdreavus tweaked the wording a little bit. let me know how that sounds

```diff
diff --git a/src/libstd/fs.rs b/src/libstd/fs.rs
index d678705626..a12f55ad9c 100644
--- a/src/libstd/fs.rs
+++ b/src/libstd/fs.rs
@@ -936,8 +936,9 @@ impl Permissions {
 
     /// Modifies the readonly flag for this set of permissions. If the
     /// `readonly` argument is `true`, using the resulting `Permission` will
-    /// make the file unwritable. Conversely, if it's is `false`, using the
-    /// resulting `Permission` will make the file writable.
+    /// update file permissions to forbid writing. Conversely, if it's `false`,
+    /// using the resulting `Permission` will update file permissions to allow
+    /// writing.
     ///
     /// This operation does **not** modify the filesystem. To modify the
     /// filesystem use the `fs::set_permissions` function.
```

**Comment by QuietMisdreavus** [comment]
Looks good, thanks!

@bors r+ rollup

**Pull request merged**